fieldservice_route/models/fsm_order.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class FSMOrder(models.Model):
+    _inherit = 'fsm.order'
+
+    dayroute_id = fields.Many2one('fsm.route.dayroute',
+                                  string='Day Route',
+                                  index=True)
+
+    def _manage_fsm_route(self, vals):
+        fsm_route_obj = self.env['fsm.route.dayroute']
+        for rec in self:
+            if not vals:
+                fsm_route = fsm_route_obj.search([
+                    ('person_id', '=', rec.person_id.id),
+                    ('date', '=', rec.scheduled_date_start.date())],
+                    limit=1)
+                rec.dayroute_id = fsm_route and fsm_route.id
+                if not fsm_route:
+                    fsm_dayroute = fsm_route_obj.create({
+                        'person_id': rec.person_id.id,
+                        'date': rec.scheduled_date_start.date(),
+                    })
+                rec.dayroute_id = fsm_dayroute.id
+            else:
+                person_id = vals.get('person_id') or rec.person_id.id
+                scheduled_date_start = vals.get('scheduled_date_start') or \
+                    rec.scheduled_date_start
+                fsm_route = fsm_route_obj.search([
+                    ('person_id', '=', person_id),
+                    ('date', '=', scheduled_date_start)],
+                    limit=1)
+                if fsm_route:
+                    vals.update({'dayroute_id': fsm_route.id})
+                elif not fsm_route and person_id and scheduled_date_start:
+                    fsm_dayroute = fsm_route_obj.create({
+                        'person_id': person_id,
+                        'date': scheduled_date_start,
+                    })
+                    vals.update({'dayroute_id': fsm_dayroute.id})
+                else:
+                    vals.update({'dayroute_id': False})
```

Comment:
@murtuzasaleh If the order is moved to another day route (date or worker is changed) or removed (date or worker are unset), the day route needs to be archived if there is no more order on it.

Try this scenario:
* Create an order with a date and worker = Worker 1 --> SR001 is created
* Edit the order to change the worker = Worker 2 --> SR002 is created, but SR001 is still active. SR001 should be archived.
* Edit the order to  change the worker = Worker 1 --> SR001 is active and has one order. SR002 is archived and is empty.